mzLib/MzLibUtil/ClassExtensions.cs

```diff
@@ -135,5 +135,70 @@ public static string GetPeriodTolerantFilenameWithoutExtension(this string fileP
             return PeriodTolerantFilenameWithoutExtension.GetPeriodTolerantFilenameWithoutExtension(filePath);
         }
 
+        /// <summary>
+        /// Determines whether the collection is null or contains no elements.
+        /// </summary>
+        /// <typeparam name="T">The IEnumerable type.</typeparam>
+        /// <param name="enumerable">The enumerable, which may be null or empty.</param>
+        /// <returns>
+        ///     <c>true</c> if the IEnumerable is null or empty; otherwise, <c>false</c>.
+        /// </returns>
+        public static bool IsNullOrEmpty<T>(this IEnumerable<T> enumerable)
```

Comment:
ICollection implements IEnumerable. You can delete this and just keep the ICollection extension

Reply:
But then we cannot use this on data types which implement IEnumerable, but not ICollection